### ISSUE TITLE
Suggest widget does not resize properly when number of suggestions increases.

### ISSUE THREAD

**Issue by alexet** [microsoft/vscode#110971]
Issue Type: <b>Bug</b>

When there is a small number of suggestions a small suggest widget is shown. If later a larger number of suggestions is given then the box remains at the small size and it isn't obvious that the rest of the suggestions gets cut off. The size seems to be based on the first box in a sequence of suggestions.

As an example here is a case using the codeql extension. Typing `m.`  in the following ends up first suggesting `m` and then tries to suggest methods on `Bar` (`foo`, `toString`, `getAQlClass`) but with a small box which only shows `foo` (although you can get the rest by scrolling.

```
newtype TFoo = Foo()
select "test"
class Bar extends Foo { predicate foo() {any()} string toString() {any()}}

predicate foo(Bar m) {
  m.
}
```

I would expect to see the same size box as I get with `Ctrl-Space` the second time around. 

It seems like all of the default providers always show many suggestions or never end up showing more suggestions so I couldn't find a case to reproduce. 

This seems to be a regression since 1.50.1 where a full sized suggestion widget was shown for the second set of suggestions

It seems to happen on macos and linux (windows not tested).

VS Code version: Code 1.51.1 (e5a624b788d92b8d34d1392e4c4d9789406efe8f, 2020-11-10T23:31:29.624Z)
OS version: Linux x64 4.19.0-12-amd64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4810MQ CPU @ 2.80GHz (8 x 3219)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: unavailable_off<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|31.02GB (13.21GB free)|
|Process Argv|--no-sandbox --unity-launch --crash-reporter-id 66c77ce2-2dd6-4bac-9c9c-633de1e518e9|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|wayland|
</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.1.13
gitlens|eam|11.0.1
tsfmt-vscode|ete|0.3.3
vscode-codeql|Git|1.3.6
vscode-test-explorer|hbe|2.19.3
vscode-graphviz|joa|0.0.6
csharp|ms-|1.23.6
python|ms-|2020.11.367453362
jupyter|ms-|2020.11.358541065
vscode-groovy-lint|Nic|1.3.0
java|red|0.71.0
rust|rus|0.7.8
vscode-lldb|vad|1.6.0
vscodeintellicode|Vis|1.2.10
vscode-java-debug|vsc|0.29.0
vscode-java-dependency|vsc|0.15.0
vscode-java-pack|vsc|0.10.0
vscode-java-test|vsc|0.26.0
vscode-maven|vsc|0.25.0

</details>

**Comment by alexet** [comment]
I have managed to make a small typescript reproducing case
```

const foobar = {
  foo: 1,
  bar: 2
};

foobar
```
Triggering suggestions at the end there gives just `foobar`, then typing `.` gives a small 1 line suggestion widget despite the fact that it contains `foo` and `bar`. 

Also the ui is particularly bad as the scrollbar is the entire height of the widget so it doesn't look like a scrollbar and hence the box doesn't look scrollable.

**Comment by dggsax** [comment]
Just came across something similar and was gonna come to see if someone had opened an issue regarding this. Just curious, @alexet  does this [StackOverflow answer](https://stackoverflow.com/a/64817846/5551875) fix your case, as in you're able to drag to increase the size of the suggestions widget?

**Comment by alexet** [comment]
It does fix it locally. I still think it is a bug as this is still the default behaviour.

**Comment by jrieken** [comment]
Did you resize the widget before? Just by itself the widget has no "custom" size and always shows 12 items. Once you have resized it the custom size will be reused.

**Comment by dggsax** [comment]
In my case, I had never resized the widget, in fact I didn't even realize it was a thing until I was looking into why I was only seeing one suggestion from intellisense. I vaguely remember reading somewhere about a bug that when the feature was released that it automatically made the widget be sized to the smallest height.

I see that there's a `Reset Suggest Widget Size` action, but didn't know about that at the time. Perhaps there can be some styling done to indicate that the widget is re-sizeable or maybe a setting that's like `suggestions-to-show` and then by default that'd be 10.

**Comment by alexet** [comment]
Running `Reset Suggest Widget Size` re-enables the bug the bug after resizing the window to prevent it from happening.

The problem is that the default widget size no-longer resizes itself bigger when new items appear after initially starting small.

A major problem is that the scroll bar is the height of a single item so it doesn't look like a scroll bar. With it looking like a scroll bar it would be much better as then at least it would not seem like a bug with missing suggestions.

**Comment by jrieken** [comment]
> The problem is that the default widget size no-longer resizes itself bigger when new items appear after initially starting small.

So, you are saying that you manually resize the widget to a smaller size and don't expect that size to be used when having different/more completions?

**Comment by alexet** [comment]
This is all using the default size after resetting the widget size

Taking the type script example we start with

```
const foobar = {
  foo: 1,
  bar: 2
};
fooba
````

I then ask for suggestions via `Ctrl-Space`. The only suggestion is `foobar` and a suggestion window with only one element is shown as expected. Then I type `.`. This updates the suggestions to now contain `foo` and `bar` but the suggestion window remain small. If I then move away and ask for suggestions again. I get the two I expected. This worked with previous versions.

Here is a gif showing the behaviour described
![Example](https://user-images.githubusercontent.com/3524095/100347825-e3240a80-2fdd-11eb-819d-b9bc65a2858f.gif)

**Comment by jrieken** [comment]
That's what I see

![Nov-26-2020 13-55-13](https://user-images.githubusercontent.com/1794099/100353618-1622ca00-2fef-11eb-9f5c-9512898baf78.gif)

**Comment by alexet** [comment]
That is what I used to see before I updated.

**Comment by jrieken** [comment]
This latest. In order to help you I need steps to reproduce. Please try again the sequence that you described and let me know if you reproduce or not

* reset suggest widget size (F1 > Reset Suggest Widget Size)
* use above sample

**Comment by misolori** [comment]
I've also encountered this several times lately. I'll try and get good repro steps, but what usually happens is I'll manually resize the widget to be a tad smaller (just a few pixels) and then layer as I intelligence comes up the widget gets resized to only show a single item and gets stuck there.

**Comment by jrieken** [comment]
> but what usually happens is I'll manually resize the widget to be a tad smaller (just a few pixels) and then later as intellisense comes up, the widget gets resized to only show a single item and gets stuck there.

In the last week I did fix a bug where you would get the "wrong" height persisted easily. This should be better now but I am curious to see if there is anything else. Tho, in all strictness what stated here (https://github.com/microsoft/vscode/issues/110971#issuecomment-734255238) is different, e.g that the widget doesn't grow with being resized

**Comment by loneshark99** [comment]
> This latest. In order to help you I need steps to reproduce. Please try again the sequence that you described and let me know if you reproduce or not
> 
>     * reset suggest widget size (F1 > Reset Suggest Widget Size)
> 
>     * use above sample

I had the same issue that the intellisense only showed 1 response in the dropdown. Struggled with this for a while and was going to move away from Vscode to  jupyter notebook, until I saw this fix after days of searching.
Thanks for the help. Makes the experience so much better.

Now sure how I got into this since on my windows box it works as expected but on Ubuntu it was not working.
